timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -593,30 +632,30 @@ def stack_distributions(distributions: List[Distribution]) -> Distribution:
             assert len(set(tuple(p.keys()) for p in params_per_dist)) == 1
 
             stacked_params = {}
-            for key in dist.arg_constraints.keys():
+            for key in last_dist.arg_constraints.keys():
                 stacked_params[key] = torch.cat([p[key] for p in params_per_dist])
-            return dist.__class__(**stacked_params)
+            return last_dist.__class__(**stacked_params)
 
-        if not isinstance(forecasts[0].distribution, AffineTransformed):
-            raise AssertionError("Expected forecast.distribution to be an instance of AffineTransformed")
+        if not isinstance(dist_forecasts[0].distribution, AffineTransformed):
+            raise AssertionError()
```

Comment:
Could this be removed given the new assertion above? https://github.com/autogluon/autogluon/pull/4969/files#diff-28a0cc813c0c8a1cd39a4e8c358ee689718114207a2655af00cb4ae7675d7665R619-R621